### PR TITLE
KFSPTS-16564 clean dashes out of ACH bank accounts before processing

### PR DESCRIPTION
**Pull request by jhulslander**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java [hunk 1]
```diff
@@ -207,6 +208,19 @@ public boolean processACHBatchDetails() {
...
+    
+    protected void cleanPayeeACHAccountExtractDetail(PayeeACHAccountExtractDetail detail) {
+        if (StringUtils.isNotBlank(detail.getBankAccountNumber()) && !StringUtils.isNumeric(detail.getBankAccountNumber())) {
+            String logMessageStarter = "cleanPayeeACHAccountExtractDetail, the bank account for " + detail.getNetID();
+            if (StringUtils.contains(detail.getBankAccountNumber(), KFSConstants.DASH)) {
+                LOG.error(logMessageStarter + " contains dashes, so removing them");
+                String cleanedBankAccount = StringUtils.remove(detail.getBankAccountNumber(), KFSConstants.DASH);
+                detail.setBankAccountNumber(cleanedBankAccount);
+            } else {
+                LOG.error(logMessageStarter + " is not numberic but does NOT contains dashes, no automatic cleaning available");
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java]: Minor typo in log message; should change from "...numberic..." to "...numeric..."

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java [hunk 2]
```diff
@@ -724,5 +724,60 @@ public void markRowAsInvalid() {
...
+    
+    @Test
+    public void testCleanPayeeACHAccountExtractDetailNoClean() {
+        String bankAccountNumber = "12345";
+        PayeeACHAccountExtractDetail detail = new PayeeACHAccountExtractDetail();
+        detail.setBankAccountNumber(bankAccountNumber);
+        payeeACHAccountExtractService.cleanPayeeACHAccountExtractDetail(detail);
+        
+        String expectedBankAccount = bankAccountNumber;
+        assertEquals(expectedBankAccount, detail.getBankAccountNumber());
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java]: These new unit test methods have a lot of duplicated code. I would suggest creating a separate helper method to do the bulk of the setup and assertion work.

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java [hunk 3]
```diff
@@ -724,5 +724,60 @@ public void markRowAsInvalid() {
...
+        PayeeACHAccountExtractDetail detail = new PayeeACHAccountExtractDetail();
+        detail.setBankAccountNumber(bankAccountNumber);
+        payeeACHAccountExtractService.cleanPayeeACHAccountExtractDetail(detail);
+        
+        String expectedBankAccount = "A123456";
+        assertEquals(expectedBankAccount, detail.getBankAccountNumber());
+    }
+    
+    @Test
+    public void testCleanPayeeACHAccountExtractDetailithLetters() {
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java]: Minor typo in method name; should be changed from "...Detailith..." to "...DetailWith..."

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java (hunk 2 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java]: good point, I shall refactor

**Comment by jhulslander** [comment]
I've made some changes, this is ready for review again.

**Pull request merged**